### PR TITLE
Add Conflict Check to OutOfOffice

### PR DESCRIPTION
**Pull request by NeilAllavarpu**

### CODE REVIEW

File: src/components/FeatureViews/Accounts/OutOfOffice.js [hunk 1]
```diff
@@ -74,167 +75,172 @@ const OutOfOffice = ({instructorID, button}) => {
...
-                                    value="primary" />
-                            </Grid>
-                            <Grid>
-                                <div className="checkboxText">
+                        md={12}>
+                        <Grid
+                            item
+                            md={3}>
+                            <div className="select">
+                                * Select OOO Start Date
```
Comment (dhuangg) [inline src/components/FeatureViews/Accounts/OutOfOffice.js]: I thought this was done a while ago, but remove the asterix, and change the field to be required

File: src/components/FeatureViews/Accounts/OutOfOffice.js [hunk 2]
```diff
@@ -74,167 +75,172 @@ const OutOfOffice = ({instructorID, button}) => {
...
+                            </div>
+                            <DatePicker
```
Comment (dhuangg) [inline src/components/FeatureViews/Accounts/OutOfOffice.js]: This should be a required field

File: src/components/FeatureViews/Accounts/OutOfOffice.js [hunk 3]
```diff
@@ -74,167 +75,172 @@ const OutOfOffice = ({instructorID, button}) => {
...
+                                onChange={setStart}
+                                openTo="day"
+                                value={start}
+                                views={["year", "month", "date"]} />
+                        </Grid>
+                        <Grid
+                            item
+                            md={3}>
+                            <div className="select">
+                                * Select OOO End Date
```
Comment (dhuangg) [inline src/components/FeatureViews/Accounts/OutOfOffice.js]: ditto

File: src/components/FeatureViews/Accounts/OutOfOffice.js [hunk 4]
```diff
@@ -74,167 +75,172 @@ const OutOfOffice = ({instructorID, button}) => {
...
+                                views={["year", "month", "date"]} />
+                        </Grid>
+                        <Grid
+                            item
+                            md={6} />
+                        <Grid
+                            item
+                            md={3}>
+                            <div className="select">
+                                * Select OOO Start Time
```
Comment (dhuangg) [inline src/components/FeatureViews/Accounts/OutOfOffice.js]: this shouldn't be a required field, remove it

File: src/components/FeatureViews/Accounts/OutOfOffice.js [hunk 5]
```diff
@@ -74,167 +75,172 @@ const OutOfOffice = ({instructorID, button}) => {
...
+                                disabled={allDay}
+                                label="Start Time"
+                                onChange={setStart}
+                                value={start} />
+                        </Grid>
+                        <Grid
+                            item
+                            md={3}>
+                            <div className="select">
+                                * Select OOO End Time
```
Comment (dhuangg) [inline src/components/FeatureViews/Accounts/OutOfOffice.js]: this shouldn't be a required field

File: src/components/FeatureViews/Accounts/OutOfOffice.js [hunk 6]
```diff
@@ -74,167 +75,172 @@ const OutOfOffice = ({instructorID, button}) => {
...
+                            md={2}>
+                            <InstructorConflictCheck
+                                ignoreAvailablity
+                                instructorID={instructorID}
+                                end={end}
+                                start={start}>
+                                <Button
+                                    className="button"
+                                    disabled={!canSubmit}>
+                                    // onClick={handleSave}>
```
Comment (dhuangg) [inline src/components/FeatureViews/Accounts/OutOfOffice.js]: Delete comment

**Comment by NeilAllavarpu** [comment]
Can you clarify what you mean by dialog contents?

**Comment by dhuangg** [comment]
anything that has a field should be wrapped in a form so that you can use the onSubmit event listener

**Comment by NeilAllavarpu** [comment]
Hmm..., that's probably a good idea, but I might have to rewrite a couple things to get that to work.

**Comment by dhuangg** [comment]
not super necessary. I'll go ahead and merge this in. I think people can just press a button lol

**Pull request merged**